src/shared/Table/ResponsiveTable.tsx

```diff
@@ -180,6 +189,31 @@ export const ResponsiveTable = <TRow, TCol>({
     [columns, getTd]
   );
 
+  const allChecked = areAllRowsChecked != undefined && areAllRowsChecked();
+
+  const isBulkCheckPropsValid = () => {
+    if (onBulkCheck != undefined && areAllRowsChecked != undefined) return true;
+    else if (
+      (onBulkCheck != undefined && areAllRowsChecked == undefined) ||
+      (onBulkCheck == undefined && areAllRowsChecked != undefined)
+    )
+      throw new Error("Not all props for checkbox have been defined");
+    return false;
+  };
+
+  const isRowCheckPropsValid = () => {
+    if (isRowChecked != undefined && onCheck != undefined) return true;
+    else if (
+      (isRowChecked != undefined && onCheck == undefined) ||
+      (isRowChecked == undefined && onCheck != undefined)
+    )
+      throw new Error("Not all props for checkbox have been defined");
```

Comment:
The messages can be more meaningful. One shouldn't have to jump into the source code to understand what the error is exactly.